feature-libs/asm/components/asm-create-customer-form/asm-create-customer-form.component.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import {
+  AsmCreateCustomerFacade,
+  CustomerRegistrationForm,
+} from '@spartacus/asm/root';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  HttpErrorModel,
+} from '@spartacus/core';
+import {
+  CustomFormValidators,
+  FocusConfig,
+  ICON_TYPE,
+  LaunchDialogService,
+} from '@spartacus/storefront';
+import { BehaviorSubject } from 'rxjs';
+import { CreatedCustomer } from './asm-create-customer-form.model';
+
+@Component({
+  selector: 'cx-asm-create-customer-form',
+  templateUrl: './asm-create-customer-form.component.html',
+})
+export class AsmCreateCustomerFormComponent {
+  createdCustomer: CreatedCustomer;
+
+  iconTypes = ICON_TYPE;
+
+  isLoading$ = new BehaviorSubject(false);
+
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: 'cx-asm-create-customer-form',
+    focusOnEscape: true,
+  };
+
+  registerForm: FormGroup = this.fb.group({
+    firstName: ['', Validators.required],
+    lastName: ['', Validators.required],
+    email: ['', [Validators.required, CustomFormValidators.emailValidator]],
+  });
+
+  constructor(
+    protected launchDialogService: LaunchDialogService,
+    protected fb: FormBuilder,
+    protected asmCreateCustomerFacade: AsmCreateCustomerFacade,
+    protected globalMessageService: GlobalMessageService
+  ) {}
+
+  submitForm(): void {
+    if (this.registerForm.valid) {
+      this.registerUser();
+    } else {
+      this.registerForm.markAllAsTouched();
+    }
+  }
+
+  registerUser(): void {
+    this.isLoading$.next(true);
+    this.setCreatedCustomer(this.registerForm.value);
+
+    this.asmCreateCustomerFacade
+      .createCustomer(this.collectDataFromRegisterForm())
+      .subscribe({
+        next: () => this.onRegisterUserSuccess(),
+        complete: () => this.isLoading$.next(false),
+        error: (error: HttpErrorModel) => this.onRegisterUserFail(error),
+      });
+  }
+
+  setCreatedCustomer(formData: any) {
+    const { firstName, lastName, email } = formData;
+    this.createdCustomer = {
+      firstName,
+      lastName,
+      email,
+    };
+  }
+  collectDataFromRegisterForm(): CustomerRegistrationForm {
+    return {
+      firstName: this.createdCustomer.firstName,
+      lastName: this.createdCustomer.lastName,
+      emailAddress: this.createdCustomer.email.toLowerCase(),
```

Comment:
Still waiting,  more information please see : https://sap-cx.slack.com/archives/C04GTG77PJN/p1677268380479409

Reply:
Keep the same behavior as the accelerator and **do not automatically lower case**,  because we do not know the root cause of the automatic conversion of case by the user registration API. We will use this ticket to track:https://jira.tools.sap/browse/CXSPA-2701, Thank you for pointing out this issue.